erpnext/hr/doctype/employee/employee.js

```diff
@@ -40,6 +40,28 @@ erpnext.hr.EmployeeController = frappe.ui.form.Controller.extend({
 				"Ms": "Female"
 			}[this.frm.doc.salutation]);
 		}
+	},
+	
+});
+frappe.ui.form.on('Employee',{
+	prefered_contact_email:function(frm){
+		frm.events.update_contact(frm)
+	},
+	personal_email:function(frm){
+		frm.events.update_contact(frm)
+	},
+	company_email:function(frm){
+		frm.events.update_contact(frm)
+	},
+	user_id:function(frm){
+		frm.events.update_contact(frm)
+	},
+	update_contact:function(frm){
+		for(var i =0, j=frm.fields.length;i<j;i++){
+			if (frm.fields[i].df.label===frm.doc.prefered_contact_email){
```

Comment:
i thought about that but it doesn't allow for customisation but if u think it should b will change